tensorflow_io/bigquery/kernels/bigquery_lib.h

```diff
@@ -50,17 +52,42 @@ Status GetDataFormat(string data_format_str,
 class BigQueryClientResource : public ResourceBase {
  public:
   explicit BigQueryClientResource(
-      std::shared_ptr<apiv1beta1::BigQueryStorage::Stub> stub)
-      : stub_(std::move(stub)) {}
-
-  std::shared_ptr<apiv1beta1::BigQueryStorage::Stub> get_stub() {
-    return stub_;
+      std::function<std::unique_ptr<apiv1beta1::BigQueryStorage::Stub>()>
+          stub_factory)
+      : stub_factory_(stub_factory) {}
+
+  explicit BigQueryClientResource()
+      : BigQueryClientResource([]() {
+          string server_name = "dns:///bigquerystorage.googleapis.com";
+          auto creds = ::grpc::GoogleDefaultCredentials();
+          grpc::ChannelArguments args;
```

Comment:
i think grpc can sometimes under the cover reuse channels with the same paremeters.  For parallelization you might try adding in a random arg to see if this can improve throughput.

Reply:
Thanks for the pointer, double-checked BigTable SDK, looks like it is doing just that: https://github.com/googleapis/google-cloud-cpp/blob/b26c708de834acda41194b241f6912f02c0fd56f/google/cloud/bigtable/internal/common_client.cc#L31
Also unfortunately this didn't improve my benchmark numbers.